scrimage-core/src/main/java/com/sksamuel/scrimage/nio/StreamingGifWriter.java

```diff
@@ -65,6 +67,10 @@ public StreamingGifWriter withInfiniteLoop(boolean infiniteLoop) {
       GifStream writeFrame(ImmutableImage image) throws IOException;
 
       GifStream writeFrame(ImmutableImage image, Duration delay) throws IOException;
+
+      GifStream writeFrame(ImmutableImage image, DisposeMethod disposeMethod) throws IOException;
+
+      GifStream writeFrame(ImmutableImage image, Duration delay, DisposeMethod disposalMethod) throws IOException;
```

Comment:
This sort of extension won't really be scalable if we allow for more metadata attributes to be set, so we could look into creating some sort of builder for this, but this is functional for now. If we add the ability to set another metadata attribute, we'll probably want to investigate alternatives.

Reply:
Yep, I might add a builder at some point for this.